fava/core/tree.py

```diff
@@ -0,0 +1,112 @@
+"""TreeNode balance trees."""
+
+import collections
+
+from beancount.core import account
+from beancount.core.amount import Amount
+from beancount.core.number import ZERO
+from beancount.core.position import Position
+
+
+#: Attributes:
+#:     name: Account name.
+#:     children: A list of :class:`TreeNode`, its children.
+#:     balance: The account balance.
+#:     balance_children: The cumulative account balance.
+#:     has_txns: Bool indicating whether this account has any transactions.
+#:
+TreeNode = collections.namedtuple(
+    'TreeNode',
+    ('name', 'children', 'balance', 'balance_children', 'has_txns'))
+
+
+# pylint: disable=abstract-method
+class CounterInventory(collections.Counter):
+    """A lightweight inventory.
+
+    This is intended as a faster alternative to Beancount's Inventory class.
+    Due to not using a list, for inventories with a lot of different positions,
+    inserting is much faster.
+
+    The keys should be tuples (currency, cost).
+    """
+
+    def is_empty(self):
+        """Check if the inventory is empty."""
+        for count in self.values():
+            if count != ZERO:
+                return False
+        return True
+
+    def reduce(self, reducer, *args):
+        """Reduce inventory.
+
+        Note that this returns a simple currency to number dict, not a
+        CounterInventory.
```

Comment:
This comment disagrees with line 47. You probably just want a dict and not a CounterInventory.

Reply:
thanks, fixed (the docstring)